pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -511,9 +511,13 @@ def _validate_monitor_key(self, trainer):
             m = (
                 f"ModelCheckpoint(monitor='{self.monitor}') not found in the returned metrics:"
                 f" {list(metrics.keys())}. "
-                f"HINT: Did you call self.log('{self.monitor}', tensor) in the LightningModule?"
             )
-            raise MisconfigurationException(m)
+            if not trainer.checkpoint_connector._one_training_epoch_completed:
+                m += "Running first epoch, a MisconfigurationException will be raise next epoch"
+                rank_zero_warn(m, UserWarning)
```

Comment:
This solution has the disadvantage that when users who forgot to log their monitor in `training_step` or `validation_step` (or mistyped their monitor), will not get a MisconfigurationError until the next epoch.

Have you thought about other ways to solve this?

Reply:
Hey @carmocca, when we forgot to log on `training_step` or `validation_step`, metrics is empty and `self._is_valid_monitor_key(metrics)` will return False, and which would skip the check.

Reply:
would it? the if condition is `... and **not** self._is_valid_monitor_key(metrics)`

Reply:
`will not get a MisconfigurationError until the next epoch.` Yes I agree with this. I guess we should introduce sanity check for train too to resolve this problem.

Reply:
> In order to resolve this, we should introduce a train sanity check, so we can capture `epoch_end hooks`

That's a good idea, try to validate the ModelCheckpoint monitors in the sanity check. Does the check not run `on_train_epoch_end` currently?